src/components/date-picker.tsx

```diff
@@ -0,0 +1,125 @@
+import { StyleSheet } from 'nativewind';
+import { type ComponentProps } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { Dropdown } from 'react-native-element-dropdown';
+import { z } from 'zod';
+
+import type { Input } from '@/ui';
+import { View } from '@/ui';
+import { white } from '@/ui/colors';
+
+type DropdownItem = {
+  label: string;
+  value: number;
+};
+
+const monthOptions = Array.from({ length: 12 }, (_, index) => ({
+  label: (index + 1).toString(),
+  value: index + 1,
+}));
+
+const yearOptions = [
```

Comment:
The yearOptions array has hardcoded values, this means you'd have to update this values every year, what if you generate it automatically instead?


```
const currentYear = new Date().getFullYear();
const yearOptions = Array.from({ length: 10 }, (_, i) => ({
  label: (currentYear + i).toString(),
  value: currentYear + i,
}));
```